examples/vision/laion5b.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from torchdata.datapipes.iter import Dropper, Filter, HuggingFaceHubReader, Mapper, Slicer
+from torchdata.datapipes.iter.load.online import _get_response_from_http
+
+try:
+    import PIL
+    from PIL import Image
+except ImportError:
+    PIL = None
+    Image = None
+
+
+def has_no_watermark(x):
+    return x["pwatermark"] is not None and x["pwatermark"] < 0.8
+
+
+def is_sfw(x):
+    return x["punsafe"] is not None and x["punsafe"] < 0.5
+
+
+def load_image(url):
+    try:
+        return _get_response_from_http(url, timeout=5)[1]
+    except Exception:
+        return None
+
+
+def image_was_loaded(x):
+    return x is not None
+
+
+# For more information about the dataset see: https://laion.ai/blog/laion-5b/
+# name of the dataset to be used
+NAME = "laion/laion2B-en-joined"
+
+
+# As the dataset is too large to store locally we use a streaming approach
+def laion2b_en(name=NAME):
+    dp = HuggingFaceHubReader(name)
```

Comment:
Noob question: Is `HuggingFaceHub` the only source of the meta data?

Reply:
Yes and no. The dataset with prejoined watermark and unsafe probabilities is only available on `HuggingFaceHub` (afaik). The dataset without this info is also available on [the-eye.eu](https://the-eye.eu/public/AI/cah/laion5b/metadata/laion2B-en/)

Reply:
Sounds good to me.